pkg/aws/aws.go

```diff
@@ -29,22 +33,55 @@ type awsSvc struct {
 }
 
 func New(configObject config.BaseConfig) (*awsSvc, error) {
+	var err error
 	if configObject.BucketName == "" {
 		return nil, fmt.Errorf("New: BucketName cannot be nil")
 	}
 
 	a := awsSvc{}
 
-	sess := session.Must(session.NewSessionWithOptions(session.Options{}))
-	a.awsSession = sess
-	a.ec2svc = ec2.New(sess, &aws.Config{
-		CredentialsChainVerboseErrors: aws.Bool(true),
-	})
+	roleArnName := os.Getenv("ROLE_ARN")
+
+	baseSess := session.Must(session.NewSessionWithOptions(session.Options{}))
+	a.awsSession = baseSess
+
+	if roleArnName != "" {
+		stsSvc := sts.New(baseSess)
+		sessionName := fmt.Sprintf("some-session-%d", rand.Int())
+		assumedRole, err := stsSvc.AssumeRole(&sts.AssumeRoleInput{
+			RoleArn: aws.String(roleArnName),
+			RoleSessionName: aws.String(sessionName),
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("Error assuming role %s", err)
+		}
+
+		assumedSession, err := session.NewSession(&aws.Config{
+			Credentials: credentials.NewStaticCredentials(
```

Comment:
Curious, instead of leveraging `AssumeRole()` + `NewStaticCredentials()` could this be instead simply done with: `creds := stscreds.NewCredentials(sess, "myRoleArn")`?

You'd just need to create the `ec2svc` with `&aws.Config{Credentials: creds}`.

https://maori.geek.nz/assuming-roles-in-aws-with-go-aeeb28fab418 is a decent example of encapsulating the creation of the client in a method.